### ISSUE TITLE
Improving Build Time

### ISSUE THREAD

**Issue by selfsame** [arcadia-unity/Arcadia#242]
Ok so I have some ideas about how to speed up build times, based on some assumptions:
* Arcadia's build finds every user .clj file in Assets and does arcadia.compiler/aot-namespaces on it
  * Unused namespaces are still built and contribute to build times
  * All user clj files in Assets must be build safe (not editor api)
  * aot-namespaces is transative, and doesn't cache what it's built, meaning every user file is a tree of compilation jobs
  * a configuration.edn :build-files ["game/core.clj"] entry could allow the user to specify targets to override this default, in many cases a single namespace has all the deps needed for the full project.

[NOTE: This proved to be untrue]
* During build, Arcadia stages the .dll files inside of Assets
  * Unity picks up on these changed libraries, the progress bar that shows up during build is Unity loading the dll assets (at that point the build has actually finished)
  * Unity then flushes the environment, incurring the lengthy Arcadia load time
  * The actual build time has always been low

**Comment by selfsame** [comment]
Related issue: if you've run `Arcadia>Compiler>AOT Compile Internal Namespaces` , building will fail due to UnityEditor references.

**Comment by selfsame** [comment]
`project.clj` files can cause issues

```
FileNotFoundException: Could not locate core.logic.project.cljc.dll or core/logic/project.cljc on load path.
```

**Comment by nasser** [comment]
Fixed by #243
